ibis/backends/base_sql/compiler.py

```diff
@@ -0,0 +1,195 @@
+from operator import add, mul, sub
+
+import ibis.backends.base_sqlalchemy.compiler as comp
+import ibis.common.exceptions as com
+import ibis.expr.datatypes as dt
+import ibis.expr.operations as ops
+import ibis.expr.types as ir
+from ibis.backends.base_sql import (
+    binary_infix_ops,
+    operation_registry,
+    quote_identifier,
+)
+
+
+def build_ast(expr, context):
+    assert context is not None, 'context is None'
+    builder = BaseQueryBuilder(expr, context=context)
+    return builder.get_result()
+
+
+def _get_query(expr, context):
+    assert context is not None, 'context is None'
+    ast = build_ast(expr, context)
+    query = ast.queries[0]
+
+    return query
+
+
+def to_sql(expr, context=None):
+    if context is None:
+        context = BaseDialect.make_context()
+    assert context is not None, 'context is None'
+    query = _get_query(expr, context)
+    return query.compile()
+
+
+# ----------------------------------------------------------------------
+# Select compilation
+
+
+class BaseSelectBuilder(comp.SelectBuilder):
+    @property
+    def _select_class(self):
+        return BaseSelect
+
+
+class BaseQueryBuilder(comp.QueryBuilder):
+
+    select_builder = BaseSelectBuilder
+
+
+class BaseContext(comp.QueryContext):
+    def _to_sql(self, expr, ctx):
+        return to_sql(expr, ctx)
+
+
+class BaseSelect(comp.Select):
+
+    """
+    A SELECT statement which, after execution, might yield back to the user a
+    table, array/list, or scalar value, depending on the expression that
+    generated it
+    """
+
+    @property
+    def translator(self):
+        return BaseExprTranslator
+
+    @property
+    def table_set_formatter(self):
+        return BaseTableSetFormatter
+
+
+class BaseTableSetFormatter(comp.TableSetFormatter):
```

Comment:
Not in this PR, but feels like it could make sense to move these few methods and attributes to their parent classes in base_sqlalchemy, and just use those. If you think this can make sense and is worth having a look, can you create an issue for it please?